### PR TITLE
[FLINK-34386][state] Add RocksDB bloom filter metrics

### PR DESCRIPTION
**Pull request by hejufang**
## What is the purpose of the change

Add RocksDB bloom filter metrics.

## Brief change log

Get RocksDB bloom filter metrics via RocksDB Statistics, and report it via Metrics reporter.

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented? not documented

### CODE REVIEW

**Review by pnowojski — COMMENTED** [review]
LGTM, but you forgot to regenerate the docs, hence the test failure. Could you fix that?

**Comment by hejufang** [comment]
@flinkbot run azure

**Comment by hejufang** [comment]
@pnowojski Thank you for your reminding. I have fixed the test failure, please review.

**Comment by hejufang** [comment]
@flinkbot run azure

**Comment by pnowojski** [comment]
@flinkbot run azure

**Comment by hejufang** [comment]
@flinkbot run azure

**Comment by hejufang** [comment]
@flinkbot  run azure

**Comment by JingGe** [comment]
There some issue while building the hadoop image: https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=57409&view=logs&j=ef799394-2d67-5ff4-b2e5-410b80c9c0af&t=9e5768bc-daae-5f5f-1861-e58617922c7a&l=10543

**Comment by JingGe** [comment]
@hejufang the issue should be fixed if you rebase your branch

**Comment by hejufang** [comment]
@flinkbot run azure

**Comment by hejufang** [comment]
@flinkbot run azure

**Comment by hejufang** [comment]
@flinkbot run azure

**Comment by hejufang** [comment]
@flinkbot run azure

**Comment by hejufang** [comment]
@JingGe Thank you for your advice. The issue is fixed.

**Comment by hejufang** [comment]
@pnowojski kindly remind.

File: flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricOptions.java [hunk 1]
```diff
@@ -262,6 +262,27 @@ public class RocksDBNativeMetricOptions implements Serializable {
...
+                    .defaultValue(false)
+                    .withDescription(
+                            "Monitor the total count of bloom FullFilter has not avoided the reads.");
+
+    public static final ConfigOption<Boolean> MONITOR_BLOOM_FILTER_FULL_TRUE_POSITIVE =
+            ConfigOptions.key("state.backend.rocksdb.metrics.bloom-filter-full-true-positive")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription(
+                            "Monitor the total count of bloom FullFilter has not avoided the reads and data actually exist.");
```
Comment (Zakelly) [inline flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricOptions.java:284]: I would suggest a refine, such as "Monitor the count of reads avoided by full filter while the data actually exists in RocksDB".

**Review by Zakelly — COMMENTED** [review]
Thanks for the PR! It looks good.

File: flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricOptions.java (hunk 1 above)
Comment (hejufang) [reply flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricOptions.java]: Thanks for your advice, I’ve refined this description.

**Comment by hejufang** [comment]
@flinkbot run azure

**Review by Zakelly — APPROVED** [review]
LGTM. Thanks for the update.

**Pull request merged**